cirq/sim/clifford/clifford_simulator.py

```diff
@@ -65,55 +68,78 @@ def is_supported_operation(op: 'cirq.Operation') -> bool:
         # TODO: support more general Pauli measurements
         return protocols.has_stabilizer_effect(op)
 
-    def _base_iterator(
-        self, circuit: circuits.Circuit, qubit_order: ops.QubitOrderOrList, initial_state: int
-    ) -> Iterator['cirq.CliffordSimulatorStepResult']:
-        """Iterator over CliffordSimulatorStepResult from Moments of a Circuit
+    def create_act_on_args(
+        self,
+        initial_state: Union[int, clifford.ActOnStabilizerCHFormArgs],
+        qubits: Sequence['cirq.Qid'],
+    ) -> clifford.ActOnStabilizerCHFormArgs:
+        """Creates the ActOnStabilizerChFormArgs for a circuit.
 
         Args:
-            circuit: The circuit to simulate.
-            qubit_order: Determines the canonical ordering of the qubits. This
-                is often used in specifying the initial state, i.e. the
-                ordering of the computational basis states.
             initial_state: The initial state for the simulation in the
                 computational basis. Represented as a big endian int.
+            qubits: Determines the canonical ordering of the qubits. This
+                is often used in specifying the initial state, i.e. the
+                ordering of the computational basis states.
 
+        Returns:
+            ActOnStabilizerChFormArgs for the circuit.
+        """
+        if isinstance(initial_state, clifford.ActOnStabilizerCHFormArgs):
+            return initial_state
+
+        qubit_map = {q: i for i, q in enumerate(qubits)}
+
+        state = CliffordState(qubit_map, initial_state=initial_state)
+        return clifford.ActOnStabilizerCHFormArgs(
+            state=state.ch_form,
+            axes=[],
+            prng=self._prng,
+            log_of_measurement_results={},
+            qubits=qubits,
+        )
+
+    def _core_iterator(
+        self,
+        circuit: circuits.Circuit,
+        sim_state: clifford.ActOnStabilizerCHFormArgs,
+    ):
+        """Iterator over CliffordSimulatorStepResult from Moments of a Circuit
+
+        Args:
+            circuit: The circuit to simulate.
+            sim_state: The initial state args for the simulation in the
+                computational basis.
 
         Yields:
             CliffordStepResult from simulating a Moment of the Circuit.
         """
-        qubits = ops.QubitOrder.as_qubit_order(qubit_order).order_for(circuit.all_qubits())
 
-        qubit_map = {q: i for i, q in enumerate(qubits)}
+        def create_state():
+            state = CliffordState(sim_state.qubit_map)
+            state.ch_form = sim_state.state.copy()
+            return state
```

Comment:
I'd recommend handling this with an optional `ch_form` argument to the `CliffordState` constructor, rather than mutating `state_ch_form` manually.